crates/typed-store/src/metrics.rs

```diff
@@ -1,12 +1,72 @@
 // Copyright (c) 2022, Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
-use std::sync::Arc;
-
 use once_cell::sync::OnceCell;
 use prometheus::{
     exponential_buckets, register_histogram_vec_with_registry,
     register_int_counter_vec_with_registry, HistogramVec, IntCounterVec, Registry,
 };
+use std::sync::atomic::{AtomicU64, Ordering};
+use std::sync::Arc;
+use std::time::Duration;
+
+#[derive(Debug, Clone)]
+// A struct for sampling based on number of operations or duration.
+// Sampling happens if the duration expires and after number of operations
+pub struct SamplingInterval {
+    // Sample once every time duration
+    pub once_every_duration: Duration,
+    // Sample once every number of operations
+    pub after_num_ops: u64,
+    // Counter for keeping track of previous sample
+    pub counter: Arc<AtomicU64>,
+}
+
+impl Default for SamplingInterval {
+    fn default() -> Self {
+        SamplingInterval::new(Duration::from_secs(1), 0)
+    }
+}
+
+impl SamplingInterval {
```

Comment:
Perhaps we should also add unit test for `once_every_duration != Duration::ZERO`